### PR TITLE
Add ci test github action

### PR DESCRIPTION
**Pull request by wemeetagain**
Probably needs more work, but here's a start.
Meant to replace the current travis functionality.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: CI Tests
+
+on: [pull_request]
```
Comment (mpetrunic) [inline .github/workflows/test.yml]: push as well?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      with:
+          cmd: run coverage
+    - name: E2e tests
+      uses: wemeetagain/actions-lerna@v0.1.0
+      with:
+          cmd: run test:e2e
+    - name: Spec tests
+      uses: wemeetagain/actions-lerna@v0.1.0
+      with:
+          cmd: run test:spec-min
```
Comment (mpetrunic) [inline .github/workflows/test.yml]: it should run full suite on master

**Review by mpetrunic — COMMENTED** [review]
any idea how to integrate docs?

**Review by mpetrunic — COMMENTED** [review]
Those parallel tests are a mess, I think running test one by one is probably cleaner. Our tests are still too short to make up for booting of virtual machine

Especially since you are running bootstrap in every single one of them haha

**Comment by wemeetagain** [comment]
might make sense to run spec tests separately and everything else as one job (lint, check-types, unit, e2e)

**Comment by mpetrunic** [comment]
> might make sense to run spec tests separately and everything else as one job (lint, check-types, unit, e2e)

maybe, but minimal spec tests are quite fast now though

File: .github/workflows/docs.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    - name: Spec tests
+      uses: wemeetagain/actions-lerna@v0.1.0
+      with:
+          cmd: run run build:release
+    - name: Build docs
+      run: node scripts/collect-docs.js
+    - name: Deploy
+      uses: peaceiris/actions-gh-pages@v2
+      env:
+        PERSONAL_TOKEN: ${{ secrets.GH_PAGES_TOKEN }}
```
Comment (priom) [inline .github/workflows/docs.yml:26]: How's this getting triggered? I don't see any variable named GH_PAGES_TOKEN in the settings.

Comment (wemeetagain) [reply .github/workflows/docs.yml:26]: you're right, I guess that was set on travis's secrets.

Comment (priom) [reply .github/workflows/docs.yml:26]: yeah so, for github actions, you have to set the secrets in the individual repo

Comment (mpetrunic) [reply .github/workflows/docs.yml:26]: can't you put secrets in repo -> settings -> secrets?

Isn't repo just for large secrets?

Comment (wemeetagain) [reply .github/workflows/docs.yml:26]: added token to repo secrets

**Comment by wemeetagain** [comment]
> https://github.com/ethereum/eth2.0-APIs/pull/14/files#diff-1d37e48f9ceff6d8030570cd36286a61R12 stuck_out_tongue
I couldn't get it to work, I think we should just merge so we can start using actions, we can incrementally improve what we have here

**Review by mpetrunic — APPROVED** [review]

**Review by twoeths — APPROVED** [review]

**Pull request merged**